File: modules/zhihu-pull-refresh/ios/PullRefreshModule.swift
import ExpoModulesCore

public final class PullRefreshModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ZhihuPullRefresh")

    View(PullRefreshView.self) {
      Events("onPull", "onRefreshTriggered")

      Prop("enabled") { (view: PullRefreshView, enabled: Bool) in
        view.setPullEnabled(enabled)
      }

      Prop("hapticsEnabled") { (view: PullRefreshView, enabled: Bool) in
        view.setHapticsEnabled(enabled)
      }

      Prop("refreshing") { (view: PullRefreshView, refreshing: Bool) in
        view.setRefreshing(refreshing)
      }

      Prop("threshold") { (view: PullRefreshView, threshold: Double) in
        view.setThreshold(threshold)
      }

      Prop("holdDistance") { (view: PullRefreshView, distance: Double) in
        view.setHoldDistance(distance)
      }

      Prop("maxPullDistance") { (view: PullRefreshView, distance: Double) in
        view.setMaxPullDistance(distance)
      }

      OnViewDestroys { view: PullRefreshView in
        view.dispose()
      }
    }
  }
}

File: modules/zhihu-pull-refresh/android/src/main/java/com/zhihuminus/pullrefresh/PullRefreshModule.kt
package com.zhihuminus.pullrefresh

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class PullRefreshModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ZhihuPullRefresh")

    View(PullRefreshView::class) {
      Events("onPull", "onRefreshTriggered")

      Prop("enabled") { view: PullRefreshView, enabled: Boolean ->
        view.setPullEnabled(enabled)
      }

      Prop("hapticsEnabled") { view: PullRefreshView, hapticsEnabled: Boolean ->
        view.setHapticsEnabled(hapticsEnabled)
      }

      Prop("refreshing") { view: PullRefreshView, refreshing: Boolean ->
        view.setRefreshing(refreshing)
      }

      Prop("threshold") { view: PullRefreshView, threshold: Float ->
        view.setThreshold(threshold)
      }

      Prop("holdDistance") { view: PullRefreshView, holdDistance: Float ->
        view.setHoldDistance(holdDistance)
      }

      Prop("maxPullDistance") { view: PullRefreshView, maxPullDistance: Float ->
        view.setMaxPullDistance(maxPullDistance)
      }

      OnViewDestroys { view: PullRefreshView ->
        view.dispose()
      }
    }
  }
}
